src/index.tsx

```diff
@@ -60,10 +67,15 @@ export const createPortalNode = <C extends Component<any>>(): PortalNode<C> => {
             }
 
             if (parent && lastPlaceholder) {
-                parent.replaceChild(
-                    lastPlaceholder,
-                    portalNode
-                );
+                // If either the PortalNode or its content is an SVG element, we need to treat it differently
+                if (portalNode instanceof SVGElement || lastPlaceholder instanceof SVGElement) {
+                    (portalNode as HTMLElement).innerHTML = '';
+                } else {
+                    parent.replaceChild(
+                        lastPlaceholder,
+                        portalNode
+                    );
```

Comment:
I think this is now addressed with the second attempt: I added this exact story (which now works properly) as well as a version where the OutPortals were rendered before the InPortals (with 50% opacity on the `<rect>`s so that they didn't obscure the text) and both are working now.

Please let me know if resolving this conversation was incorrect, or if I've missed anything!